src/amm/erc_4626/mod.rs

```diff
@@ -369,17 +392,18 @@ mod tests {
     #[tokio::test]
     async fn test_calculate_price() -> eyre::Result<()> {
         let rpc_endpoint = std::env::var("ETHEREUM_RPC_ENDPOINT")?;
-        let middleware = Arc::new(Provider::<Http>::try_from(rpc_endpoint)?);
+        let provider =
+            Arc::new(ProviderBuilder::<_, _, AnyNetwork>::default().on_http(rpc_endpoint.parse()?));
 
         let mut vault = ERC4626Vault {
-            vault_token: H160::from_str("0x163538E22F4d38c1eb21B79939f3d2ee274198Ff")?,
+            vault_token: address!("163538E22F4d38c1eb21B79939f3d2ee274198Ff"),
             ..Default::default()
         };
 
-        vault.populate_data(None, middleware).await?;
+        vault.populate_data(None, provider).await?;
 
-        vault.vault_reserve = U256::from_dec_str("501910315708981197269904")?;
-        vault.asset_reserve = U256::from_dec_str("505434849031054568651911")?;
+        vault.vault_reserve = U256::from_str("501910315708981197269904")?;
+        vault.asset_reserve = U256::from_str("505434849031054568651911")?;
```

Comment:
Nit: I believe we should just be able to use `U256::from` here. Same thing throughout the tests with similar cases.